### ISSUE TITLE
Task from 133: Add error handling for HSES data

### ISSUE THREAD

**Issue by jduss4** [OHS-Hosting-Infrastructure/complaint-tracker#135]
This task is part of #133 : incorporating HSES staging data in the complaint tracker

This task is scoped to:
- [ ] handling different types of errors in response from HSES gracefully
- [ ] logging response
- [ ] providing the user with friendly information

We will require design for potential messages to replace "Placeholder text for the message, including any mitigation steps we need to share," which is currently what displays for no results.

**Comment by lauranash18f** [comment]
To make sure I am tracking the design needs, you need an empty state for the table?

**Comment by jduss4** [comment]
There are a couple potential things we can expect back from HSES:

- 200 (yay it works)
- 400 Bad request
- 401 Authorization information is invalid
- 403 Remote IP is not allowed
- 404 Not found
- 500 Unexpected error

Some of these are likely our problems, like 400, 401, and 403.  For something like 404, what kind of a message should we display to the user if they have attempted to view an issue that HSES doesn't think exists?

Also, what kind of generic "check back, having trouble getting info from HSES" type of error would we like to display to the user for a 500?

**Comment by lauranash18f** [comment]
Thanks! I can map these to messages — are there any steps a user can realistically take for any of these, other than checking back later?

As for the 500, we have this from the field-level errors that I can use as a MadLib:
We're unable to look up reports in TTA Smart Hub right now. Please try again later.

We're unable to _______ in _____ right now. Please try again later.

**Comment by jduss4** [comment]
The only action I think I user could take for a 404 is if they are manually altering the URL to try to get to a specific issue and mistyped, which seems like an edge case.  It still might be nice for them to know that the issue is "not found" rather than just a generic error, though?  What do you think?

I like the 500 error pattern!

**Comment by lauranash18f** [comment]
Agreed, I usually like being as specific as possible with errors (within reason of course!) so that the person has a clue about what's happening (and if it's different from what happened before). 

When do you need these by? They should go quick but want to prioritize them appropriately.

**Comment by jduss4** [comment]
Would Wednesday be too soon?

**Comment by lauranash18f** [comment]
I can do Wednesday! If I can't get to it Tuesday, I have time
before training starts — so that's realistic. Please include it as a
blocker in standup if I am, in fact, blocking you or about to block you as
a friendly nudge.
